RoadNamer/Tools/RoadSelectTool.cs

```diff
@@ -33,35 +40,52 @@ protected override void OnEnable()
         protected override void OnDisable()
         {
             base.OnDisable();
+            EventBusManager.Instance().Publish("closeAll", null);
         }
 
         protected override void OnToolUpdate()
         {
             if (m_toolController != null && !m_toolController.IsInsideUI && Cursor.visible)
             {
-                RaycastOutput raycastOutput;
+                netSegmentIds.Clear();
 
+                RaycastOutput raycastOutput;
                 if (RaycastRoad(out raycastOutput))
                 {
                     ushort netSegmentId = raycastOutput.m_netSegment;
 
                     if (netSegmentId != 0)
                     {
-                        NetManager netManager = Singleton<NetManager>.instance;
-                        NetSegment netSegment = netManager.m_segments.m_buffer[(int)netSegmentId];
+
+                        NetSegment netSegment = netManager.m_segments.m_buffer[netSegmentId];
+
+                        netSegmentIds.Add(netSegmentId);
+                        if (Input.GetKey(KeyCode.LeftShift))
+                        {
+                            List<ushort> connectedSegments = getConnectingSegments(netSegmentId);
+                            connectedSegments.Remove(netSegmentId);
+                            netSegmentIds.AddRange(connectedSegments);
+                            if (m_roadNamePanel != null)
+                            {
+                                m_roadNamePanel.m_netSegmentIds = new List<ushort>(netSegmentIds);
+                            }
+                        }
 
                         if (netSegment.m_flags.IsFlagSet(NetSegment.Flags.Created))
                         {
-                            if (Event.current.type == EventType.MouseDown /*&& Event.current.button == (int)UIMouseButton.Left*/)
+                            if (Event.current.type == EventType.MouseDown && Event.current.button == (int)UIMouseButton.None)
```

Comment:
Should add a comment here ( decompiled values for left and right button clicks are 0 and 1, which are defined as None/Left for some reason )